restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/IncomingPhoneNumbersEndpoint.java

```diff
@@ -654,4 +650,108 @@ public Response deleteIncomingPhoneNumber(final String accountSid, final String
+                    return status(NOT_FOUND).entity("Destination organization not found").build();
+                }
+                migrateNumbersFromAccountTree(targetAccount, destinationOrganization);
+                return status(OK).build();
+            }
+        }catch(Exception e){
+            logger.error("Exception while performing migrateIncomingPhoneNumbers: ", e);
+            return status(INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    /**
+     * @param targetAccount
+     * @param destinationOrganization
+     */
+    private void migrateNumbersFromAccountTree(Account targetAccount, Organization destinationOrganization) {
+        List<String> subAccountsToClose = accountsDao.getSubAccountSidsRecursive(targetAccount.getSid());
+        if (subAccountsToClose != null && !subAccountsToClose.isEmpty()) {
+            int i = subAccountsToClose.size(); // is is the count of accounts left to process
+            // we iterate backwards to handle child account numbers first, parent account's numbers next
+            while (i > 0) {
+                i --;
+                String migrateSid = subAccountsToClose.get(i);
+                try {
+                    Account subAccount = accountsDao.getAccount(new Sid(migrateSid));
+                    migrateSingleAccountNumbers(subAccount, destinationOrganization);
+                } catch (Exception e) {
+                    // if anything bad happens, log the error and continue migrating the rest of the accounts numbers.
+                    logger.error("Failed migrating (child) account's numbers: account sid '" + migrateSid + "'");
+                }
+            }
+        }
+        // migrate parent account numbers too
+        migrateSingleAccountNumbers(targetAccount, destinationOrganization);
+
+    }
+
+    /**
+     * @param account
+     * @param destinationOrganization
+     */
+    private void migrateSingleAccountNumbers(Account account, Organization destinationOrganization) {
+        List<IncomingPhoneNumber> incomingPhoneNumbers = dao.getIncomingPhoneNumbers(account.getSid());
+        if(incomingPhoneNumbers != null) {
+            for (int i=0; i<incomingPhoneNumbers.size(); i++){
+                IncomingPhoneNumber incomingPhoneNumber = incomingPhoneNumbers.get(i);
+                incomingPhoneNumber.setOrganizationSid(destinationOrganization.getSid());
+                //update organization in db
+                dao.updateIncomingPhoneNumber(incomingPhoneNumber);
+                //update number at provider's end
+                if(!updateNumberAtPhoneNumberProvisioningManager(incomingPhoneNumber, destinationOrganization)){
+                    //if number could not be updated at provider's end, log the error and keep moving to next number.
+                    logger.error(String.format("could not update numner %s at number provider %s ", incomingPhoneNumber.getPhoneNumber(), phoneNumberProvisioningManager));
```

Comment:
maybe error level is too much. INFO or WARN would be enough i think...

Reply:
since this part of API is for admins, error would be something they would be interested to look at in logentries etc